DependencyInjection/DoctrineExtension.php

```diff
@@ -254,6 +255,54 @@ protected function getConnectionOptions($connection)
     {
         $options = $connection;
 
+        $options['connection_override_options'] = [];
+
+        $connectionDefaults = [
+            'dbname' => null,
+            'host' => 'localhost',
+            'port' => null,
+            'user' => 'root',
+            'password' => null,
+        ];
+
+        $allowOverrideUrl = $options['override_url'] ?? false;
+
+        if (! $allowOverrideUrl && isset($options['url'])) {
+            Deprecation::trigger(
+                'doctrine/doctrineBundle',
+                'https://github.com/doctrine/DoctrineBundle/pull/1290',
+                'Not setting doctrine.dbal.override_url to true is deprecated. True is the only value that will be supported in doctrine-bundle 3.0.'
+            );
+        }
+
+        if ($allowOverrideUrl) {
+            foreach (array_keys($connectionDefaults) as $paramName) {
+                if (! array_key_exists($paramName, $options)) {
+                    continue;
+                }
+
+                $options['connection_override_options'][$paramName] = $options[$paramName];
+            }
+        }
+
+        unset($options['override_url'], $connectionDefaults['dbname']);
+
+        $nestedConnections = ['shards', 'replicas', 'slaves'];
+
+        foreach ($connectionDefaults as $defaultKey => $defaultValue) {
```

Comment:
Follow up to our slack conversation and to bring everyone else up to speed:

If we set default values in the `Configuration` - we have no way to determine if a value is an override or a default. https://github.com/doctrine/DoctrineBundle/pull/1290/files#diff-59ba6fdc91d16d6c28504e469b134f4bbf84e6b39d42c83d91652afe6015e1e7 asserts this behavior. If we set the defaults in the `Configuration`, this test will now fail(`host` would be `localhost` rather than the overridden value).